osidb/serializer.py

```diff
@@ -436,32 +435,41 @@ class BugzillaSyncMixinSerializer(serializers.ModelSerializer):
     which need to perform Bugzilla sync as part of the save procedure
     """
 
-    bz_api_key = serializers.CharField(allow_null=False, required=True, write_only=True)
+    def __get_bz_api_key(self):
+        bz_api_key = self.context["request"].META.get("HTTP_BUGZILLA_API_KEY")
+        if not bz_api_key:
+            raise serializers.ValidationError(
+                {"Bugzilla-Api-Key": "This HTTP header is required."}
+            )
+        return bz_api_key
 
     def create(self, validated_data):
         """
         perform the ordinary instance create
         with providing BZ API key while saving
         """
-        bz_api_key = validated_data.pop("bz_api_key")
+        # NOTE: This won't work for many-to-many fields as
+        # some logic from original .create() was overwritten
+
         instance = self.Meta.model(**validated_data)
-        instance.save(bz_api_key=bz_api_key)
+        instance.save(bz_api_key=self.__get_bz_api_key())
         return instance
 
     def update(self, instance, validated_data):
         """
         perform the ordinary instance update
         with providing BZ API key while saving
         """
-        bz_api_key = validated_data.pop("bz_api_key")
+        # NOTE: This won't work for many-to-many fields as
+        # some logic from original .create() was overwritten
         for attr, value in validated_data.items():
             setattr(instance, attr, value)
-        instance.save(bz_api_key=bz_api_key)
+        instance.save(bz_api_key=self.__get_bz_api_key())
         return instance
 
     class Meta:
         model = BugzillaSyncMixin
-        fields = ["bz_api_key"]
+        fields = []
```

Comment:
I wonder whether this is even necessary now. May we should rather remove the lines with
```
            + BugzillaSyncMixinSerializer.Meta.fields
```